src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -165,14 +165,68 @@ IntDoubleStr ToNumericValueGetter::operator()(
   AD_FAIL();
 }
 
-// ____________________________________________________________________________
+// _____________________________________________________________________________
 IntDoubleStr ToNumericValueGetter::operator()(
     const LiteralOrIri& s,
     [[maybe_unused]] const EvaluationContext* context) const {
   return std::string(asStringViewUnsafe(s.getContent()));
 }
 
-// ____________________________________________________________________________
+// _____________________________________________________________________________
+OptIri DatatypeValueGetter::operator()(ValueId id,
+                                       const EvaluationContext* context) const {
+  using enum Datatype;
+  auto datatype = id.getDatatype();
+  std::optional<std::string> entity;
+  switch (datatype) {
+    case Bool:
+      return Iri::fromIrirefWithoutBrackets(XSD_BOOLEAN_TYPE);
+    case Double:
+      return Iri::fromIrirefWithoutBrackets(XSD_DOUBLE_TYPE);
+    case Int:
+      return Iri::fromIrirefWithoutBrackets(XSD_INT_TYPE);
+    case Date: {
+      auto dateType = id.getDate().toStringAndType().second;
+      if (dateType != nullptr) {
+        return Iri::fromIrirefWithoutBrackets(dateType);
+      } else {
+        return std::nullopt;
+      }
+    }
+    case LocalVocabIndex:
+    case VocabIndex:
+      return (*this)(ExportQueryExecutionTrees::getLiteralOrIriFromVocabIndex(
+                         context->_qec.getIndex(), id, context->_localVocab),
+                     context);
```

Comment:
I think in some of the SparqlExpressionTestHelperWhatevers files I already have a prepared
EvaluationContext which has an index and a local vocab with some IDs. That you can reuse in your tests to also 
make these two changes here covered.